src/gluonts/transform/sampler.py

```diff
@@ -55,31 +56,64 @@ class UniformSplitSampler(InstanceSampler):
     """
 
     @validated()
-    def __init__(self, p: float) -> None:
+    def __init__(self, p: float, **kwargs) -> None:
+        super().__init__(**kwargs)
         self.p = p
 
-    def __call__(self, ts: np.ndarray, a: int, b: int) -> np.ndarray:
-        assert (
-            a <= b
-        ), "First index must be less than or equal to the last index."
+    def __call__(self, ts: np.ndarray) -> np.ndarray:
+        a, b = self._get_bounds(ts)
+
+        if a > b:
+            return np.array([], dtype=int)
 
         window_size = b - a + 1
         (indices,) = np.where(np.random.random_sample(window_size) < self.p)
         return indices + a
 
 
-class TestSplitSampler(InstanceSampler):
+class PredictionSplitSampler(InstanceSampler):
     """
     Sampler used for prediction. Always selects the last time point for
     splitting i.e. the forecast point for the time series.
     """
 
     @validated()
-    def __init__(self) -> None:
-        pass
-
-    def __call__(self, ts: np.ndarray, a: int, b: int) -> np.ndarray:
-        return np.array([b])
+    def __init__(
+        self,
+        allow_empty_interval=False,
+        axis: int = -1,
+        min_past: int = 0,
+        min_future: int = 0,
+    ) -> None:
+        super().__init__(axis=axis, min_past=min_past, min_future=min_future)
```

Comment:
Can't we just convert this into `pydantic` types?

Reply:
done, it looks much better